.github/workflows/spotless-check.yml

```diff
@@ -0,0 +1,43 @@
+name: Spotless Check
+
+on: [pull_request]
+
+jobs:
+  spotless:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@17d0e2bd7d51742c71671bd19fa12bdc9d40a3d6 # v2.8.1
+        with:
+          egress-policy: audit
+
+      - name: Checkout code
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          fetch-depth: 0
+
```

Comment:
Do we also want to expand the git clone to ensure spotless can apply to all of the code?
```yaml
      - name: Expand Shallow Clone for Spotless
        run: git fetch --unshallow --no-recurse-submodules
```

Reply:
tried to add that step but got the following error, so reverted back to how it was:

```
Run git fetch --unshallow --no-recurse-submodules
  git fetch --unshallow --no-recurse-submodules
  shell: /usr/bin/bash -e {0}
fatal: --unshallow on a complete repository does not make sense
Error: Process completed with exit code 1[2](https://github.com/hashgraph/hedera-block-node/actions/runs/9898941525/job/27346849024?pr=43#step:5:2)8.
```

Reply:
Upon further research, realized that I can put a conditional to unshallow only if necessary.
![image](https://github.com/user-attachments/assets/a929f528-e40e-455e-a7af-c0200f013120)